src/transformers/models/univnet/modeling_univnet.py

```diff
@@ -0,0 +1,677 @@
+            num_blocks=kernel_predictor_num_blocks,
+            resnet_in_channels=cond_channels,
+            resnet_hidden_channels=kernel_predictor_hidden_channels,
+            resnet_kernel_size=kernel_predictor_conv_size,
+            dropout=kernel_predictor_dropout,
+            leaky_relu_slope=kernel_predictor_leaky_relu_slope,
+        )
+
+        self.resblocks = nn.ModuleList(
+            [
+                UnivNetLVCResidualBlock(
+                    hidden_channels=hidden_channels,
+                    kernel_size=kernel_size,
+                    dilation=dilations[i],
+                    leaky_relu_slope=resnet_leaky_relu_slope,
+                )
+                for i in range(len(dilations))
+            ]
+        )
+
+    def forward(self, hidden_states: torch.FloatTensor, spectrogram: torch.FloatTensor):
+        # hidden_states: (batch_size, hidden_channels, seq_length)
+        # spectrogram: (batch_size, cond_channels, cond_length)
+        hidden_states = nn.functional.leaky_relu(hidden_states, self.leaky_relu_slope)
+        hidden_states = self.convt_pre(hidden_states)
+
+        kernels, biases = self.kernel_predictor(spectrogram)
+
+        for i, resblock in enumerate(self.resblocks):
+            kernel = kernels[:, i, :, :, :, :]
+            bias = biases[:, i, :, :]
+            hidden_states = resblock(hidden_states, kernel, bias, hop_size=self.cond_hop_length)
+
+        return hidden_states
+
+    def apply_weight_norm(self):
+        nn.utils.weight_norm(self.convt_pre)
+        self.kernel_predictor.apply_weight_norm()
+        for layer in self.resblocks:
+            layer.apply_weight_norm()
+
+    def remove_weight_norm(self):
+        nn.utils.remove_weight_norm(self.convt_pre)
+        self.kernel_predictor.remove_weight_norm()
+        for layer in self.resblocks:
+            layer.remove_weight_norm()
+
+
+UNIVNET_START_DOCSTRING = r"""
+    This model inherits from [`PreTrainedModel`]. Check the superclass documentation for the generic methods the
+    library implements for all its model (such as downloading or saving, resizing the input embeddings, pruning heads
+    etc.)
+
+    This model is also a PyTorch [torch.nn.Module](https://pytorch.org/docs/stable/nn.html#torch.nn.Module) subclass.
+    Use it as a regular PyTorch Module and refer to the PyTorch documentation for all matter related to general usage
+    and behavior.
+
+    Parameters:
+        config ([`UnivNetGanConfig`]):
+            Model configuration class with all the parameters of the model. Initializing with a config file does not
+            load the weights associated with the model, only the configuration. Check out the
+            [`~PreTrainedModel.from_pretrained`] method to load the model weights.
+"""
+
+
+@add_start_docstrings(
+    """UnivNet GAN vocoder.""",
+    UNIVNET_START_DOCSTRING,
+)
+class UnivNetGan(PreTrainedModel):
+    config_class = UnivNetGanConfig
+    main_input_name = "spectrogram"
```

Comment:
if you want this to work with accelerate you are probably gonna need a `no_split_module` attribute!

Reply:
Would you be able to provide some documentation and/or examples? I'm not super familiar with `accelerate`.

Reply:
Essentially, the `_no_split_module` attribute tells `accelerate` which submodules share weights (e.g., embedding layers in the input and output embeddings). These modules should **not** end up in different FSDP wrapped units. There's a bit of documentation on this page if you search for `_no_split_module`: https://huggingface.co/docs/transformers/main_classes/trainer

I think this model is ok without a `_no_split_module` attribute since there are no shared weights.